google/resource_bigquery_dataset.go

```diff
@@ -184,6 +184,15 @@ func resourceBigQueryDataset() *schema.Resource {
 				},
 			},
 
+			// Delete Contents on Destroy: [Optional] If True, delete all the tables in the dataset.
+			// If False and the dataset contains tables, the request will fail.
+			// Default is False.
+			"delete_contents_on_destroy": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
```

Comment:
I'm inclined to remove the default; as-is, Terraform will report a diff for every existing user of bigquery dataset, even if it's a no-op. We can fix that by either removing the default (it will actually still default to false, it just won't be recorded in state) & presumably all the `ImportStateVerify` checks but the one where the value is explicitly set.

Alternatively, we could add a `DiffSuppressFunc` either similar to or reusing https://github.com/terraform-providers/terraform-provider-google/blob/69e445bd59301e491e17d3490babc6e2262ebf08/google/utils.go#L154 (Judging by the `SchemaDiffSuppressFunc`, Terraform treats booleans as strings inside that function)

Reply:
I think I like the latter option better, it makes for more readable code :) Thanks! I didn't know of the existence of this function haha.